crates/types/src/test_helpers.rs

```diff
@@ -36,3 +71,603 @@ pub fn create_contract<R: Rng>(
+    )
+        .prop_map(|(policies, inputs, outputs, witnesses, payload)| {
+            let blob_id = BlobId::compute(&payload);
+            let body = BlobBody {
+                id: blob_id,
+                witness_index: 0,
+            };
+            let blob = crate::fuel_tx::Transaction::blob(
+                body, policies, inputs, outputs, witnesses,
+            );
+            Transaction::Blob(blob)
+        })
+}
+
+prop_compose! {
+    fn arb_consensus_header()(
+        prev_root in any::<[u8; 32]>(),
+        time in any::<u64>(),
+    ) -> crate::blockchain::header::ConsensusHeader<GeneratedConsensusFields> {
+        crate::blockchain::header::ConsensusHeader {
+            prev_root:  prev_root.into(),
+            height: BlockHeight::new(0),
+            time: Tai64(time),
+            generated: GeneratedConsensusFields::default(),
+        }
+    }
+}
+
+prop_compose! {
+    /// Generate an arbitrary block with a variable number of transactions
+    pub fn arb_block()(
+        txs in arb_txs(),
+        da_height in any::<u64>(),
+        consensus_parameter_version in any::<u32>(),
+        state_transition_bytecode_version in any::<u32>(),
+        msg_ids in arb_msg_ids(),
+        event_root in any::<[u8; 32]>(),
+        mut consensus_header in arb_consensus_header(),
+    ) -> (Block, Vec<MessageId>, Bytes32) {
+        let mut fuel_block = Block::default();
+
+        *fuel_block.transactions_mut() = txs;
+
+        fuel_block.header_mut().set_da_height(DaBlockHeight(da_height));
+        fuel_block.header_mut().set_consensus_parameters_version(consensus_parameter_version);
+        fuel_block.header_mut().set_state_transition_bytecode_version(state_transition_bytecode_version);
+
+        let count = fuel_block.transactions().len().try_into().expect("we shouldn't have more than u16::MAX transactions");
+        let msg_root = msg_ids
+            .iter()
+            .fold(MerkleRootCalculator::new(), |mut tree, id| {
+                tree.push(id.as_ref());
+                tree
+            })
+            .root()
+            .into();
+        let tx_root = generate_txns_root(fuel_block.transactions());
+        let event_root = event_root.into();
+        fuel_block.header_mut().set_transactions_count(count);
+        fuel_block.header_mut().set_message_receipt_count(msg_ids.len().try_into().expect("we shouldn't have more than u32::MAX messages"));
+        fuel_block.header_mut().set_transaction_root(tx_root);
+        fuel_block.header_mut().set_message_outbox_root(msg_root);
+        fuel_block.header_mut().set_event_inbox_root(event_root);
+
+        // Consensus
+        // TODO: Include V2 Application with V2 Header
+        let application_hash = fuel_block.header().application_v1().unwrap().hash();
+        consensus_header.generated.application_hash = application_hash;
+        fuel_block.header_mut().set_consensus_header(consensus_header);
+        (fuel_block, msg_ids, event_root)
```

Comment:
`Block::new` should be enough for you to create a header and a block. You don't need to implement all the logic by yourself.

You can clean up getters and setters which you've added, after you udpated code here=)